apps/explorer/src/components/module/module-functions-interaction/FuncitonParamInput.tsx

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import type { FieldPath, FieldValues, UseFormRegister } from 'react-hook-form';
+
+type FunctionParamInputProps<T extends FieldValues, N extends FieldPath<T>> = {
+    paramTypeTxt: string;
+    paramIndex: number;
+    register: UseFormRegister<T>;
+    id: string;
+    name: N;
+};
+
+export function FunctionParamInput<
```

Comment:
IMO you don't need this level of type safety here, you can just use wide types. This is already going to be somewhat loose because it's an API driven form.